data-prepper-plugins/mutate-event-processors/src/main/java/org/opensearch/dataprepper/plugins/processor/mutateevent/AddEntryProcessorConfig.java

```diff
@@ -54,12 +58,20 @@ public boolean hasValueOrFormat() {
         }
 
         public Entry(final String key,
+                     final String metadataKey,
                      final Object value,
                      final String format,
                      final boolean overwriteIfKeyExists,
                      final String addWhen)
         {
+            if (key != null && metadataKey != null) {
+                throw new IllegalArgumentException("Only one of the two - key and metadatakey - should be specified");
```

Comment:
I don't feel too strongly about it but is there any reason that we can
t set both at once? Seems like there may be a case where a value should be set in both metadata and the Event

Reply:
@graytaylor0 , do you foresee a scenario where a user wants the same data in the event and metadata. With the current proposal a user can do that with:

```
  processor:
    - add_entries:
        entries:
        - key: "newMessage"
           format: "new ${message}"
           overwrite_if_key_exists: true
        - metadata_key: "newMessage"
           format: "new ${message}"
           overwrite_if_key_exists: true
```
I like this because it's explicit and follows our exiting model of one entry equaling one mutation.

Reply:
May be we can add that if needed. For now, this is not supported.

Reply:
Seems fine either way